cstrml/market/src/lib.rs

```diff
@@ -581,17 +583,50 @@ impl<T: Trait> Module<T> {
             });
 
             // 4. Add OrderSlashRecord
-            let slash_record = ProviderPunishment {
+            let provider_punishment = ProviderPunishment {
                 success: 0,
                 failed: 0,
                 value: Zero::zero()
             };
-            <ProviderPunishments<T>>::insert(&order_id, slash_record);
+            <ProviderPunishments<T>>::insert(&order_id, provider_punishment);
 
             true
         }
     }
 
+    fn maybe_close_sorder(order_id: &<T as system::Trait>::Hash) {
+        let so = Self::storage_orders(order_id).unwrap();
+        let punishment = Self::provider_punishments(order_id).unwrap();
+        if punishment.value < so.amount {
```

Comment:
AFAIK, it should not just `return`, cause the punishment should only affect punishment's closing logic?